dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h

```diff
@@ -307,7 +292,7 @@ class DMFilePackFilter
     RSCheckParam param;
 
     std::vector<RSResult> handle_res;
-    std::vector<UInt8> use_packs;
+    std::vector<bool> use_packs;
```

Comment:
I think `boost::dynamic_bitset` is a better choice when you really need it.

> If the size of the bitset is known at compile time, [std::bitset](https://en.cppreference.com/w/cpp/utility/bitset) may be used, which offers a richer set of member functions. In addition, [boost::dynamic_bitset](https://www.boost.org/doc/libs/release/libs/dynamic_bitset/dynamic_bitset.html) exists as an alternative to std::vector<bool>.

Reply:
```cpp
namespace boost {

template <typename Block, typename Allocator>
class dynamic_bitset
{
public:

    class [reference](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#reference)
    {
        void operator&(); // not defined
    public:
        // An automatically generated copy constructor.

        reference& operator=(bool value);
        reference& operator=(const reference& rhs);

        reference& operator|=(bool value);
        reference& operator&=(bool value);
        reference& operator^=(bool value);
        reference& operator-=(bool value);

        bool operator~() const;
        operator bool() const;
        reference& flip();
    };
    typedef bool [const_reference](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#const_reference);


    reference [operator[]](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#bracket)(size_type pos);
    bool [operator[]](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#const-bracket)(size_type pos) const;
};
```

seems that dynamic_bitset operator[] also return the reference 😭

Reply:
https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html

Reply:
@JaySon-Huang `dynamic_bitset` make iter inconvenient, so keep `vector<UInt8>`.